llvm/include/llvm/SandboxIR/Utils.h

```diff
@@ -49,6 +49,16 @@ class Utils {
     return const_cast<Instruction *>(I);
   }
 
+  /// \Returns the base Value for load or store instruction \p LSI.
+  template <typename LoadOrStoreT>
+  static Value *getMemInstructionBase(const LoadOrStoreT *LSI) {
+    static_assert(std::is_same_v<LoadOrStoreT, LoadInst> ||
+                      std::is_same_v<LoadOrStoreT, StoreInst>,
+                  "Expected sandboxir::Load or sandboxir::Store!");
+    return LSI->Ctx.getOrCreateValue(
```

Comment:
Do we need `getOrCreateValue()` ? I assume that sandbox IR would have been generated for the whole function. That would remove the need for `friend class Utils` in Context.
```
auto *V = LSI->Ctx.getValue( getUnderlyingObject(LSI->getPointerOperand()->Val));
assert(V != nullptr && "Missing Sandbox IR value!");
return V;
```

Reply:
getUnderlyingObject returns an llvm::Value, getOrCreateValue converts that to a sandboxir::Value.

Reply:
Yes, but there is also a public `sandboxir::Context::getValue(llvm::Value *)` that converts LLVM IR to Sandbox IR